src/CraftingRecipes.cpp

```diff
@@ -287,6 +287,60 @@ cCraftingRecipes::~cCraftingRecipes()
 
 
 
+bool cCraftingRecipes::doWeKnowAllIngredients(cRecipe * Recipe, int ItemID, std::map<int, bool> KnownItems)
```

Comment:
Refrain from using "we" in function names.
`areAllIngredientsKnown` is a better name.
Style: arguments should have the `a_` (or `a`) prefix.
`KnownItems` should be a const reference, otherwise calling this function creates a needless copy.